pkg/github/github.go

```diff
@@ -146,6 +158,53 @@ func NewClient(ctx context.Context, token string, opts ...Option) *GitHubClient
 	return g
 }
 
+// ListRepositories all repositories and returns details about the repositories.
+func (g *GitHubClient) ListRepositories(ctx context.Context, owner string, opts *github.RepositoryListByOrgOptions) ([]*Repository, error) {
+	pageStart := func(i *int) bool { return i == nil }
+	pageEnd := func(i *int) bool { return *i == 0 }
+	uniqueResponses := make(map[int64]*Repository)
+	opt := *opts
+
+	var page *int // Use nil to indicate start of request.
+
+	for pageStart(page) || !pageEnd(page) {
+		if err := g.withRetries(ctx, func(ctx context.Context) error {
+			if page != nil {
+				opt.Page = *page
+			}
+			repos, resp, err := g.client.Repositories.ListByOrg(ctx, owner, &opt)
+			if err != nil {
+				if _, ok := ignoredStatusCodes[resp.StatusCode]; !ok {
+					return retry.RetryableError(err)
+				}
+
+				return fmt.Errorf("failed to list issues: %w", err)
+			}
+
+			for _, r := range repos {
```

Comment:
Whats the use case where we get duplicate repos?

Reply:
https://github.com/abcxyz/guardian/pull/134#discussion_r1265598622

I will add a comment to justify this.

Reply:
got it, thanks!